neurots/generate/tree.py

```diff
@@ -248,8 +248,23 @@ def next_point(self):
         for section_grower in ordered_sections:
             # the current section_grower is generated
             # In here the stop criterion can be modified accordingly
+
+            if (
+                section_grower.process == "major" and "cut_major_length" in self.params
+            ):  # pragma: no cover
+                # this makes an early termination of major branch
+                # it needs to revert the value after getting the state to preserve
+                # the original topology
+                _term = section_grower.stop_criteria["TMD"].term
+                section_grower.stop_criteria["TMD"].term = self.params["cut_major_length"]
+            else:
+                _term = None
+
             state = self.growth_algo.extend(section_grower)
 
+            if _term is not None:  # pragma: no cover
```

Comment:
I'm not sure I understand this hack. Why not add it to the if / else statement?

Reply:
because the previous line is outside the if. The second if is to revert the .term change, so that the algo is not affected, but only the .extend function

Reply:
@lidakanari , let me know if you think of a better implementation, otherwise we can merge and move on to the other PR.